### DISCUSSION TITLE
Cancel or edit changes from Sverchok fork

### DISCUSSION THREAD

**Discussion by rendetto** [nortikin/sverchok#4870] [Q&A] [answered]
How to cancel or edit the changes to the code I created in a branch in my fork of Sverchok?
I didn't created a Pull Request still and want to change the description of my changes - is that possible?
Is it also possible to remove those changes?

Or I can just create another change with correct description and then do the PR?

Thanks

**Comment by portnov** [comment, marked as answer]
You mean you want to edit commit descriptions?
If you need to edit the description of only the last commit, you can do `git commit --ammend`. Text editor will open and let you to edit your message.
If you need to edit a number of messages in your branch, you can do `git rebase --interactive <STARTING_POINT>`, for example `git rebase --interactive master`, if you started your branch from master. Text editor will open with a list of commits you've made from that starting point, prepended with the word "pick". This is actually a list of commands, to apply changes in this order. You can change the word "pick" to "reword", where you want to edit the message. After you close text editor, it will open again letting you to edit messages of the commits you've marked with "reword".
With `git rebase --interactive/-i` you can do other interesting things, for example squash several commits into one, split one commit into several, or just remove some commits form history. See `git help rebase`.

The important thing to remember is that if you've edited something in your branch's history, if you have already pushed it somewhere (for example to github), when you will push it again you will have to do `git push -f`, to say that you are not only adding new commits, but rewriting the history.

**Reply by rendetto** [reply]
Thanks a lot! I'll study those options. Meanwhile an interesting thing happened - the patch I did disappeared, after an hour me doing nothing with it.
